lib/core/client.js

```diff
@@ -343,6 +343,15 @@ class Client extends EventEmitter {
 
     resume(this)
   }
+
+  set tlsSession (session) {
+    this[kTLSSession] = session
+    this.emit(kTLSSession, session)
+  }
+
+  get tlsSession () {
+    return this[kTLSSession]
+  }
```

Comment:
I'm not too happy about making these part of the public API.

Reply:
Can't pool just pass in sessions or pass a connect method through the constructor?

Reply:
Thanks for the comment @ronag and agree, this should not be part of the public API. This PR is a draft and I'm reviewing and making final tweaks, including this one.